File: src/main/kotlin/com/hdj/backend_assignment/service/PatientService.java
package com.hdj.backend_assignment.service;

import com.hdj.backend_assignment.web.dto.PatientRequestDTO;
import com.hdj.backend_assignment.web.dto.PatientResponseDTO;
import org.jetbrains.annotations.NotNull;
import org.springframework.stereotype.Service;

@Service
public class PatientService {
    @NotNull
    public PatientResponseDTO.RegisterResultDTO registerPatient(@NotNull PatientRequestDTO.RegisterDTO request) {
        return null;
    }
}

File: src/main/kotlin/com/hdj/backend_assignment/service/PatientService.kt
package com.hdj.backend_assignment.service

import com.hdj.backend_assignment.web.dto.PatientRequestDTO
import com.hdj.backend_assignment.web.dto.PatientResponseDTO

interface PatientService {

    fun registerPatient(request: PatientRequestDTO.RegisterDTO): PatientResponseDTO.RegisterResultDTO

    fun getPatient(patientId: Long): PatientResponseDTO.PatientDTO

    fun updatePatient(patientId: Long, request: PatientRequestDTO.UpdateDTO): PatientResponseDTO.PatientDTO

    fun deletePatient(patientId: Long)

    fun searchPatient(request: PatientRequestDTO.SearchDTO): PatientResponseDTO.PatientListDTO
}
